dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -338,6 +340,22 @@ public Contentlet findContentletByIdentifier(String identifier, boolean live, lo
 
     }
 
+    public List<Contentlet> findPagesByTemplate(Template template, User user, boolean respectFrontendRoles)
+            throws DotDataException, DotSecurityException {
+        try {
+            return permissionAPI.filterCollection(
+                    search("+_all:" + template.getIdentifier() + " +baseType:"
+                                    + BaseContentType.HTMLPAGE.getType(), -1, 0, null, user,
+                            respectFrontendRoles),
+                    PermissionAPI.PERMISSION_READ,
+                    respectFrontendRoles,
+                    user);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)